docs/tasks.md

```diff
@@ -872,6 +872,66 @@ log into the `Pod` and add a `Step` that pauses the `Task` at the desired stage.
 
 ```
 
+### Running Step Containers as a Non Root User
+
+All steps that do not require to be run as a root user should make use of TaskRun features to 
+designate the container for a step runs as a user without root permissions. As a best practice, 
+running containers as non root should be built into the container image to avoid any possibility 
+of the container being run as root. However, as a further measure of enforcing this practice, 
+steps can make use of a `securityContext` to specify how the container should run.
+
+An example of running Task steps as a non root user is shown below:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: show-non-root-steps
+spec:
+  steps:
+    # no securityContext specified so will use 
+    # securityContext from TaskRun podTemplate
+    - name: show-user-1001
+      image: ubuntu
+      command:
+        - ps
+      args:
+        - "aux"
+    # securityContext specified so will run as  
+    # user 2000 instead of 1001
+    - name: show-user-2000
+      image: ubuntu
+      command:
+        - ps
+      args:
+        - "aux"
+      securityContext:
+        runAsUser: 2000
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  generateName: show-non-root-steps-run-
+spec:
+  taskRef:
+    name: show-non-root-steps
+  podTemplate:
+    securityContext:
+      runAsNonRoot: true
+      runAsUser: 1001
+```
+
+In the example above, the step `show-user-2000` specifies via a `securityContext` that the container 
+for the step should run as user 2000. A `securityContext` must still be specified via a TaskRun `podTemplate` 
```

Comment:
I'm wary of redocumenting things that kubernetes does but do you think it would it be worth calling out `runAsNonRoot` for what it does?  It is suuuuper confusing that it looks so similar to runAsUser but only validates.

Reply:
Yeah, I agree about it being confusing. I can tweak the language to highlight this.

Reply:
Added